From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Fri, 24 Feb 2017 19:29:33 +0100
Subject: [PATCH] {MAYBE UPS} drivers/scsi/aacraid: fix
 -Wsometimes-uninitialized

clang reports:
    drivers/scsi/aacraid/linit.c:798:14: error: variable 'command' is
    used uninitialized whenever 'if' condition is false
    [-Werror,-Wsometimes-uninitialized]
                    } else if (aac->hba_map[bus][cid].reset_state >= 1) {
                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/scsi/aacraid/linit.c:819:25: note: uninitialized use occurs
    here
                    status = aac_hba_send(command, fib,
                                          ^~~~~~~
    drivers/scsi/aacraid/linit.c:798:10: note: remove the 'if' if its
    condition is always true
                    } else if (aac->hba_map[bus][cid].reset_state >= 1) {
                           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    drivers/scsi/aacraid/linit.c:768:13: note: initialize the variable
    'command' to silence this warning
                    u8 command;
                              ^
                               = '\0'
---
 drivers/scsi/aacraid/linit.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/aacraid/linit.c b/drivers/scsi/aacraid/linit.c
index 050954ae6095..ea1e71102d23 100644
--- a/drivers/scsi/aacraid/linit.c
+++ b/drivers/scsi/aacraid/linit.c
@@ -795,7 +795,7 @@ static int aac_eh_reset(struct scsi_cmnd* cmd)
 
 			command = HBA_IU_TYPE_SCSI_TM_REQ;
 			aac->hba_map[bus][cid].reset_state++;
-		} else if (aac->hba_map[bus][cid].reset_state >= 1) {
+		} else /* if (aac->hba_map[bus][cid].reset_state >= 1) TODO: can it be <0? */ {
 			struct aac_hba_reset_req *rst;
 
 			/* already tried, start a hard reset now */
-- 
